packages/edit-site/src/components/page-pages/index.js

```diff
@@ -266,6 +278,20 @@ export default function PagePages() {
 		totalPages,
 	} = useEntityRecords( 'postType', postType, queryArgs );
 
+	useEffect( () => {
```

Comment:
Isn't this a duplicate of the effect above? Also the code feels a bit fragile to me to be honest. Do we expect all dataviews to have this same logic. It's not really straightforward. Is there a way to improve this?

Reply:
It is not the same effect as above. Above, when the page gets a postId in the URL parameter, we set that ID as something to be part of the selection. We cannot select it immediately because the items may still be loading.
In this effect, if things are loaded and an item should be selected, we update the selection.
It looks fragile but seems to work well. Previously, I tried other ideas, but they ended up not working all the time or being even more complex. I am not sure how we can simplify this further. What I can do is put the effects and the state inside a hook that we can reuse in other places if needed (and that way, at least the logic is joined in a specific place)?

Reply:
I made this change to extract the logic in a specific hook.